File: Provider/ContentProvider/app/src/main/java/com/example/contentprovider/database/UserDAO.java
package com.example.contentprovider.database;

import com.example.contentprovider.model.User;

import java.util.List;

public interface UserDAO {
    void insertUser(User user);

    //Update name,address base on ID
    void updateUser(User user);

    boolean isUserExist(User user);

    void deleteUser(User user);

    List<User> getListUser();

    void deleteAll();
}

File: SQLite/app/src/main/java/com/example/sqlite/database/UserDAO.kt
package com.example.sqlite.database

import com.example.sqlite.model.User

interface UserDAO {

    fun insertUser(user: User)

    //Update name,address base on ID
    fun updateUser(user: User)

    fun isUserExist(user: User): Boolean

    fun deleteUser(user: User)

    fun getListUser(): List<User>

    fun deleteAll()
}